cmd/vicadmin/server.go

```diff
@@ -551,6 +548,10 @@ func (s *server) index(res http.ResponseWriter, req *http.Request) {
 	defer trace.End(trace.Begin(""))
 	ctx := context.Background()
 	sess, err := s.getSessionFromRequest(ctx, req)
+	if err != nil {
+		http.Redirect(res, req, "/logout", http.StatusTemporaryRedirect)
+		return
```

Comment:
Sidenote - while I was implementing the "degraded" dashboard behavior, I ran into this exact redirect issue upon redirecting the user to the logout handler if the session was nil.